src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -122,7 +126,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     List<Card> allCards;
     try {
       allCards =
-          executor.invokeAll(cardCallbacks, 3L, TimeUnit.SECONDS).stream()
+          executor.invokeAll(cardCallbacks, 30L, TimeUnit.SECONDS).stream()
```

Comment:
how was this timeout decided?

Reply:
Uh not all the cards were coming up all the time after adding the query to datastore so the timing was bumped up.

Reply:
I'd really prefer to have a shorter timeout. No one waits 30 seconds for a page to load. Isn't this an artifact from when we hypothesized that pages were timing out, but was fixed by adding the background thread factory?